core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/dml.scala

```diff
@@ -38,13 +38,52 @@ case class UpdateTable(
   override def output: Seq[Attribute] = target.output
 }
 
-case class MergeTables(
+/**
+ * The logical plan of the MERGE INTO command, aligned with SparkSQL
+ */
+case class MergeIntoTable(
     target: LogicalPlan,
-    source: Option[LogicalPlan],
-    conditions: Option[Expression],
+    source: LogicalPlan,
+    mergeCondition: Expression,
+    matchedActions: Seq[MergeAction],
+    notMatchedActions: Seq[MergeAction],
+    notMatchedBySourceActions: Seq[MergeAction],
     outputRelation: Option[LogicalPlan],
     options: Option[Expression])
     extends Modification {
-  override def children: Seq[LogicalPlan] = Seq(target, source.getOrElse(NoopNode), outputRelation.getOrElse(NoopNode))
+
+  override def children: Seq[LogicalPlan] = {
+    val notMatchedValues = notMatchedActions.collect { case insertAction: InsertAction =>
+      insertAction.values
+    }
+    Seq(target, source, outputRelation.getOrElse(NoopNode)) ++ notMatchedValues
+  }
   override def output: Seq[Attribute] = target.output
 }
+
+sealed abstract class MergeAction extends Expression {
+  def condition: Option[Expression]
+  override def dataType: DataType = UnresolvedType
+  override def children: Seq[Expression] = condition.toSeq
+}
+
+case class DeleteAction(condition: Option[Expression]) extends MergeAction
+
+case class UpdateAction(condition: Option[Expression], assignments: Seq[Expression]) extends MergeAction {
+  override def children: Seq[Expression] = condition.toSeq ++ assignments
+}
+
+case class UpdateStarAction(condition: Option[Expression]) extends MergeAction {
```

Comment:
we neither have `UpdateStarAction`, nor `InsertStarAction`. Rewrite the code
<img width="606" alt="image" src="https://github.com/user-attachments/assets/f098ab73-c428-43f0-aed3-2bce54d86620">

Reply:
Left in from Catalyst, which seems to special case them. They can just be removed.

Reply:
As in, you wish to special case them like SparkSql does?

Reply:
As TSQL does not support INSERT * anyway, I have just removed them. But please let me know what our Catalyst reference repo is.